lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp

```diff
@@ -0,0 +1,719 @@
+    mock_read_spgw_ue_state_db(name_of_ue_samples);
+
+    std::this_thread::sleep_for(
+        std::chrono::milliseconds(END_OF_TEST_SLEEP_MS));
+    std::cout << "Setup done" << std::endl;
+  }
+
+  virtual void TearDown() {
+    send_terminate_message_fatal(&task_zmq_ctx_main_spgw);
+    destroy_task_context(&task_zmq_ctx_main_spgw);
+    itti_free_desc_threads();
+
+    free_mme_config(&mme_config);
+    free_spgw_config(&spgw_config);
+
+    // Sleep to ensure that messages are received and contexts are released
+    std::this_thread::sleep_for(
+        std::chrono::milliseconds(END_OF_TEST_SLEEP_MS));
+  }
+
+ protected:
+  std::shared_ptr<MockMmeAppHandler> mme_app_handler;
+  std::string test_imsi_str = "001010000000002";
+  uint64_t test_imsi64      = 1010000000002;
+  uint64_t test_imsi64_test = 1010000000001;
+
+  plmn_t test_plmn = {.mcc_digit2 = 0,
+                      .mcc_digit1 = 0,
+                      .mnc_digit3 = 0x00,
+                      .mcc_digit3 = 0,
+                      .mnc_digit2 = 0,
+                      .mnc_digit1 = 0};
+  bearer_context_to_be_created_t sample_default_bearer_context = {
+      .eps_bearer_id    = 5,
+      .bearer_level_qos = {
+          .pci = 1,
+          .pl  = 15,
+          .qci = 9,
+          .gbr = {},
+      }};
+
+  bearer_qos_t sample_dedicated_bearer_qos = {.pci = 1,
+                                              .pl  = 15,
+                                              .qci = 9,
+                                              .gbr = {}};
+
+  int test_mme_s11_teid = 4;
+  int test_bearer_index = 5;
+  in_addr_t test_ue_ip  = 0x05030201;
+  in_addr_t test_ue_ip2 = 0x0d80a8c0;
+  std::vector<std::string> name_of_ue_samples;
+};
+
+TEST_F(SPGWAppInjectedStateProcedureTest, TestDeleteSessionSuccess) {
+  spgw_state_t* spgw_state = get_spgw_state(false);
+
+  spgw_ue_context_t* ue_context_p = spgw_get_ue_context(test_imsi64);
+  teid_t ue_sgw_teid =
+      LIST_FIRST(&ue_context_p->sgw_s11_teid_list)->sgw_s11_teid;
+
+  s_plus_p_gw_eps_bearer_context_information_t* spgw_eps_bearer_ctxt_info_p =
+      sgw_cm_get_spgw_context(ue_sgw_teid);
+
+  sgw_eps_bearer_ctxt_t* eps_bearer_ctxt_p = sgw_cm_get_eps_bearer_entry(
+      &spgw_eps_bearer_ctxt_info_p->sgw_eps_bearer_context_information
+           .pdn_connection,
+      DEFAULT_EPS_BEARER_ID);
+
+  ASSERT_TRUE(eps_bearer_ctxt_p->paa.ipv4_address.s_addr == test_ue_ip2);
+
+  // create sample modify default bearer request
+  itti_s11_modify_bearer_request_t sample_modify_bearer_req = {};
+  fill_modify_bearer_request(
+      &sample_modify_bearer_req, DEFAULT_MME_S11_TEID, ue_sgw_teid,
+      DEFAULT_ENB_GTP_TEID, DEFAULT_BEARER_INDEX, DEFAULT_EPS_BEARER_ID);
+
+  EXPECT_CALL(*mme_app_handler, mme_app_handle_modify_bearer_rsp()).Times(1);
+  status_code_e return_code =
+      sgw_handle_modify_bearer_request(&sample_modify_bearer_req, test_imsi64);
+
+  ASSERT_EQ(return_code, RETURNok);
+
```

Comment:
Since the UE is attached, there will not be any modify bearer request. Please remove this.